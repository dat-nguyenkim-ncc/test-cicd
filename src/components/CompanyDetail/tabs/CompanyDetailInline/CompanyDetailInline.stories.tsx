import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CompanyDetailInline, { CompanyDetailInlineProps } from './CompanyDetailInline'
import { Section } from '../../../primitives'

export default {
  title: 'Company Details/CompanyDetailInline',
  component: CompanyDetailInline,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<CompanyDetailInlineProps> = args => (
  <Section sx={{ m: 6, mx: 'auto' }}>
    <CompanyDetailInline {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {
  title: '',
  detail: '',
} as CompanyDetailInlineProps
