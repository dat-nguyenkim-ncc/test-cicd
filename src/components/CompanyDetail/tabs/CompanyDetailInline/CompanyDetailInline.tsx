import React from 'react'
import { Flex } from 'theme-ui'
import { ViewInterface } from '../../../../types'
import { Paragraph } from '../../../primitives'

export type CompanyDetailInlineProps = ViewInterface<{
  title: string
  detail: string
}>

const CompanyDetailInline = ({ title, detail, sx }: CompanyDetailInlineProps) => {
  return (
    <Flex key={title} sx={{ pr: 2, mb: 4 }}>
      <Paragraph bold sx={{ mr: 1 }}>
        {title}
      </Paragraph>
      <div className="tooltip">
        <Paragraph
          sx={{
            maxWidth: 200,
            whiteSpace: 'nowrap',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            ...sx,
          }}
        >
          {detail}
        </Paragraph>
        {detail.length > 25 && <span className="tooltiptext">{detail}</span>}
      </div>
      <style>
        {`
          .tooltiptext {
            visibility: hidden;
            background-color: black;
            color: #fff;
            border-radius: 6px;
            padding: 5px;
          
            /* Position the tooltip */
            position: absolute;
            z-index: 1;
            margin-top: 5px;
            max-width: 30%;
            word-break: break-all;
          }
          
          .tooltip:hover .tooltiptext {
            visibility: visible;
          }`}
      </style>
    </Flex>
  )
}

export default CompanyDetailInline
