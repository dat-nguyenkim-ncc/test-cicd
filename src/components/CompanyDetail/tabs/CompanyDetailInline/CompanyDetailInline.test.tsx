import * as React from 'react'
import { shallow } from 'enzyme'
import CompanyDetailInline from './CompanyDetailInline'

describe('CompanyDetailInline', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<CompanyDetailInline title="FUNDING TOTAL" detail="Series E+" />)
    expect(wrapper).toMatchSnapshot()
  })
})
