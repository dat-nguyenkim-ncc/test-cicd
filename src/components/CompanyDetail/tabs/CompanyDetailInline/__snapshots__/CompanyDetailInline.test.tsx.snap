// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetailInline matches snapshot 1`] = `ShallowWrapper {}`;
